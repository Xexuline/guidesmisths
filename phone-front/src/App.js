// import logo from './logo.svg';
import './App.scss';
import Header from './components/header/header'
import Footer from './components/footer/footer'
import Layout from './components/layout/layout'

function App() {
  return (
    <div className="App">
      <Layout 
        header={ <Header></Header> }
        footer={ <Footer></Footer> }
        left={ <div>left</div> }
        right={ <div>rigth</div> }
        >
        <p>hola</p>
      </Layout>
    </div>
  );
}

export default App;
