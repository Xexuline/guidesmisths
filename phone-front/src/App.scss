@import './styles/base/utils';

.App {
  text-align: center;
  min-width: px2rem(375);

  &-logo {
    height: 40vmin;
    pointer-events: none;
  }

  &__wrapper {
    max-width: px2rem(1200);
    display: flex;
    flex-direction: column;
  }
}
