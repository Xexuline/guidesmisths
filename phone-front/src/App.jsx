// import logo from './logo.svg';
import React, { Component } from 'react'
import './App.scss';
import Header from './components/header/header'
import Footer from './components/footer/footer'
import Layout from './components/layout/layout'
import PhonesContainer from './containers/phonesContainer/phonesContainer'
import DescriptionContainer from './containers/descriptonContainer/descriptionContainer'
import Modal from './components/modal/modal'
import Spinner from './components/loader/loader'
import { PhoneService } from './services/phone'
import { BrowserRouter, Switch, Route } from 'react-router-dom'

class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      loading: false,
      phoneList: []
    }

    this.getPhonesList = this.getPhonesList.bind(this)
  }

  getPhonesList() {
    this.setState({ loading: true })
    PhoneService.getList()
      .then(({data: phoneList}) => {
        this.setState({ 
          loading: false,
          phoneList
        })
      })
  }

  componentDidMount() {
    this.getPhonesList()
  }

  lockBody(loading){
    const body = document.querySelector('body')
    // debugger
    body.classList[loading ? 'add' : 'remove']('modal-open')
  }

  componentDidUpdate(prevProp, prevState) {
    // debugger
    if(prevState.loading !== this.state.loading) {
      this.lockBody(this.state.loading)
    }
  }

  render() {
    // const classNames = ['App', this.state.loading ? 'modal-open' : null].join(' ')
    return (
      <BrowserRouter>
        <div className='App'>
          <div className="App__wrapper">
            <Header />
            <Switch>
              <Route path="/" exact={ true } component={ () => <PhonesContainer phones={ this.state.phoneList } /> }></Route>
              <Route path="/new" exact={ true } component={ () => <div>create new</div>}></Route>
              <Route path="/update/:id" exact={ true } component={ () => <div>create new</div>}></Route>
              <Route path="/:id" component={ () => <div>con id</div> }></Route>
            </Switch>
            { this.state.loading ? <Modal><Spinner /></Modal> : null }
          </div>
        </div>
      </BrowserRouter>
    );
  }
}
// function App() {

//   function changeSelected() {
//     alert('2')
//   }
  
//   return (
//     <div className="App">
//       <Layout 
//         header={ <Header /> }
//         footer={ <Footer /> }
//         left={ <PhonesContainer selectPhone={ changeSelected }/> }
//         right={ <DescriptionContainer/> }
//         >
//       </Layout>
//     </div>
//   );
// }

export default App;
