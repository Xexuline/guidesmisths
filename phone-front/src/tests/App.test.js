import { render } from '@testing-library/react';
import App from '../App';
import React from 'react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux'
 
const mockStore = configureStore([]);
const history = createMemoryHistory()

describe('base app test', () => {
  let store
  let component
  beforeEach(() => {
    store = mockStore({
      ui: {},
    })
    component = render(
      <Provider store={store}>
        <Router history={ history }>
          <App />
        </Router>
      </Provider>
    );
  })

  it('should render', () => {
    expect(component).toMatchSnapshot()
  });
})
