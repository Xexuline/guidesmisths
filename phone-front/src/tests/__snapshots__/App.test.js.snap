// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base app test should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <main
          class="App__wrapper"
        >
          <header
            class="header"
          >
            <button
              class="icon icon-arrow header__button"
            />
            <h1>
              Phone catalog
            </h1>
            <button
              class="icon icon-plus header__button"
            />
          </header>
          <article
            class="phone-container__wrapper"
          >
            <h2>
               Phone List
            </h2>
            <div
              class="phone-container"
            />
            <button
              class="icon icon-arrow back-to-top "
            />
          </article>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <main
        class="App__wrapper"
      >
        <header
          class="header"
        >
          <button
            class="icon icon-arrow header__button"
          />
          <h1>
            Phone catalog
          </h1>
          <button
            class="icon icon-plus header__button"
          />
        </header>
        <article
          class="phone-container__wrapper"
        >
          <h2>
             Phone List
          </h2>
          <div
            class="phone-container"
          />
          <button
            class="icon icon-arrow back-to-top "
          />
        </article>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
